cmd/otel-allocator/watcher/file.go

```diff
@@ -17,56 +17,66 @@ package watcher
 import (
 	"path/filepath"
 
+	promconfig "github.com/prometheus/prometheus/config"
+
 	"github.com/fsnotify/fsnotify"
 	"github.com/go-logr/logr"
 
 	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/config"
 )
 
 type FileWatcher struct {
+	logger         logr.Logger
 	configFilePath string
 	watcher        *fsnotify.Watcher
 	closer         chan bool
 }
 
-func newConfigMapWatcher(logger logr.Logger, config config.CLIConfig) (FileWatcher, error) {
+func NewFileWatcher(logger logr.Logger, config config.CLIConfig) (Watcher, error) {
 	fileWatcher, err := fsnotify.NewWatcher()
 	if err != nil {
 		logger.Error(err, "Can't start the watcher")
-		return FileWatcher{}, err
+		return &FileWatcher{}, err
 	}
 
-	return FileWatcher{
+	return &FileWatcher{
+		logger:         logger,
```

Comment:
Would it be helpful to give this logger a name `logger.WithName("allocator-file-watcher")` or is this the callers responsibility?

Reply:
In this case its the caller's responsibility